app/api/utility.py

```diff
@@ -196,29 +212,55 @@ def create_query(
                 ?dataset_uuid nb:hasPortalURI ?dataset_portal_uri.
             }}
             OPTIONAL {{
-                ?subject nb:hasAge ?age.
+                ?session nb:hasAge ?age.
             }}
             OPTIONAL {{
-                ?subject nb:hasSex ?sex.
+                ?session nb:hasSex ?sex.
             }}
             OPTIONAL {{
-                ?subject nb:hasDiagnosis ?diagnosis.
+                ?session nb:hasDiagnosis ?diagnosis.
             }}
             OPTIONAL {{
-                ?subject nb:isSubjectGroup ?subject_group.
+                ?session nb:isSubjectGroup ?subject_group.
             }}
             OPTIONAL {{
-                ?subject nb:hasAssessment ?assessment.
+                ?session nb:hasAssessment ?assessment.
+            }}
+            {{
+                SELECT ?subject (count(distinct ?phenotypic_session) as ?num_phenotypic_sessions)
+                WHERE {{
+                    ?subject a nb:Subject.
+                    OPTIONAL {{
+                        ?subject nb:hasSession ?phenotypic_session.
+                        ?phenotypic_session a nb:PhenotypicSession.
+                        OPTIONAL {{
+                            ?phenotypic_session nb:hasAge ?age.
+                        }}
+                        OPTIONAL {{
+                            ?phenotypic_session nb:hasSex ?sex.
+                        }}
+                        OPTIONAL {{
+                            ?phenoypic_session nb:hasDiagnosis ?diagnosis.
+                        }}
+                        OPTIONAL {{
+                            ?phenotypic_session nb:isSubjectGroup ?subject_group.
+                        }}
+                        OPTIONAL {{
+                            ?phenotypic_session nb:hasAssessment ?assessment.
+                        }}
+                    }}
+                    {phenotypic_session_level_filters}
+                }} GROUP BY ?subject
```

Comment:
I think this is cleaner to wrap my head around than sticking both pheno and imaging sub-searches in the same `SELECT` statement - but just for my understanding: is that the only reason you made them into two separate sub-queries? Or is there some other aspect that makes this necessary and that I am missing?

Reply:
Yes, I thought that two sub-queries (and sets of query filters) was more intuitive and easier to read.

From what I can tell, you can achieve functionally the same behaviour with a single `SELECT` statement!